src/main/scala/uk/gov/hmrc/initprototype/PackageLockReportTask.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.initprototype
+
+import java.io.{File, PrintWriter}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.io.Source
+import scala.util.{Success, Try}
+
+case class PackageLockReportTask(
+  downloader: GithubRepositoryDownloader,
+  conf: PackageLockReportConfiguration
+) {
+
+  def checkoutAndReport(args: Array[String]): Future[Unit] = Future {
+    val directory  = ensurePrototypesDirectory(conf.prototypesDirectory)
+    val herokuRows = parseHerokuUsageReport()
```

Comment:
@JoPintoPaul not essential, but what do you think about in the variable names rather than having herokuRows instead having herokuApps - it took me a moment to work out what a heroku row was when reading for the first time - total optional thing though